portfolio/src/main/webapp/script.js

```diff
@@ -13,27 +13,45 @@
 // limitations under the License.
 
 /**
- * @param array, a non-empty array
- * @return a random element of the input array
+ * Fetches current user from UserInformationServlet and stores it in getCurrentUser.currentUser
+ * If the user is not logged in the fetch will have a 400 response code, and the user stored will be null
  */
-function getRandomElement(array) {
-  console.assert(array.length==0, "getRandomElement cannot be used on empty array");
-  return array[Math.floor(Math.random() * array.length)];
+async function initGetCurrentUser() {
+  response = await fetch('/userInfo');
+  if(!response.ok){
+    getCurrentUser.currentUser = null;
+    getCurrentUser.initalized = true;
+  } else {
+    user = await response.json();
+    getCurrentUser.currentUser = user;
+    getCurrentUser.initalized = true;
+  }
 }
 
 /**
- * Adds a random greeting to the page.
+ * Returns the Current Logged in user object
+ * If no user is logged in, return null
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+async function getCurrentUser() {
+  if(getCurrentUser.initalized === true){
```

Comment:
The first time that GetCurrentUser is called, do we know what getCurrentUser.initalized is?  I assume it would be something like Nothing?  Is that true.  Should we be initializing it before reading it the first time?

Reply:
Its undefined meaning there has been no value assigned to it, but this evaluates to null with type correction.

Reply:
In my opinion, this is hard behavior for other people who are reading your code to infer.  Is there a better way that you think that we should do this?

Reply:
I did some more digging and the seemingly obvious solution is a good option. JS has undefined as a keyword, so doing something like `(getCurrentUser.cachedUser === undefined)` is valid.